xmrstak/backend/cpu/minethd.cpp

```diff
@@ -84,7 +84,7 @@ bool minethd::thd_setaffinity(std::thread::native_handle_type h, uint64_t cpu_id
 	CPU_ZERO(&mn);
 	CPU_SET(cpu_id, &mn);
 	return pthread_setaffinity_np(h, sizeof(cpuset_t), &mn) == 0;
-#else
+#elif !defined(__OpenBSD__)
```

Comment:
Could you please try to change line 62 to 
`#elif defined(__FreeBSD__) || defined(__OpenBSD__)`
and than remove this line. This should solve #771 without removing the thread pinning feature.

Reply:
I did, however it produces:
```
/home/vtamara/comp/monero/xmr-stak/xmrstak/backend/cpu/minethd.cpp:88:2: error:
      unknown type name 'cpu_set_t'
        cpu_set_t mn;
        ^
/home/vtamara/comp/monero/xmr-stak/xmrstak/backend/cpu/minethd.cpp:91:42: error:
      use of undeclared identifier 'cpu_set_t'
        return pthread_setaffinity_np(h, sizeof(cpu_set_t)...
                                                ^
2 errors generated.
```

What I have found about CPU affinity in OpenBSD up to now are these commits from 2009:
<https://cvsweb.openbsd.org/cgi-bin/cvsweb/src/sys/sys/proc.h.diff?r1=1.112&r2=1.113&f=h>
<https://cvsweb.openbsd.org/cgi-bin/cvsweb/src/sys/sys/proc.h.diff?r1=1.115&r2=1.116&f=h>
they introduce the structure `struct cpuset` and some operations on them, however up to OpenBSD 6.2 there is no function `pthread_setaffinity_np`.

Reply:
OK in this case could you please add a extra code path for OPENBSD and show the user a WARNING that thread pinning is not supported under OPENBSD.
```
printer::inst()->print_msg(L0,"WARNING: thread pinning is not supported under OPENBSD.");
```

Reply:
Done